app/code/Magento/Theme/view/frontend/templates/html/header.phtml

```diff
@@ -6,19 +6,20 @@
 
 /**
  * @var \Magento\Theme\Block\Html\Header $block
+ * @var \Magento\Framework\Escaper $escaper
  */
 $welcomeMessage = $block->getWelcome();
 ?>
 <?php if ($block->getShowPart() == 'welcome') : ?>
     <li class="greet welcome" data-bind="scope: 'customer'">
         <!-- ko if: customer().fullname  -->
         <span class="logged-in"
-              data-bind="text: new String('<?= $block->escapeHtml(__('Welcome, %1!', '%1')) ?>').replace('%1', customer().fullname)">
+              data-bind="text: new String('<?= $escaper->escapeHtml(__('Welcome, %1!', '%1')) ?>').replace('%1', customer().fullname)">
         </span>
         <!-- /ko -->
         <!-- ko ifnot: customer().fullname  -->
         <span class="not-logged-in"
-              data-bind='html:"<?= $block->escapeHtml($welcomeMessage) ?>"'></span>
+              data-bind="html: "<?= $escaper->escapeHtml($welcomeMessage) ?>"></span>
```

Comment:
This solution looks unreliable. Please tru to leave quotes as was but change escaper method to escapeHtmlAttribute
Please add tests to cover changes